ext/opcache/jit/zend_jit_ir.c

```diff
@@ -10064,6 +10064,22 @@ static int zend_jit_do_fcall(zend_jit_ctx *jit, const zend_op *opline, const zen
 				ir_STORE(jit_EX(opline), jit_IP(jit));
 			}
 			jit_observer_fcall_begin(jit, rx, observer_handler);
+
+			if (trace) {
+				int32_t exit_point = zend_jit_trace_get_exit_point(opline, ZEND_JIT_EXIT_TO_VM);
+
+				exit_addr = zend_jit_trace_get_exit_addr(exit_point);
+				if (!exit_addr) {
+					return 0;
+				}
+			} else {
+				exit_addr = NULL;
+			}
+
+			if (!zend_jit_check_timeout(jit, NULL /* we're inside the called function */, exit_addr)) {
+				return 0;
+			}
+
```

Comment:
How this is related to the API?

Reply:
> Note, this will prevent JIT-ing of the function, but not the execution of the previously JIT-ed traces that are started in different functions and inline (or return to) the blacklisted one.

Yes, that's how this is related. It's an escape hatch how I am manually able to force-skip manually disabled functions by returning `EG(vm_interrupt)` in an observer. I have no better idea on how to achieve that short of keeping a mapping on what functions are inlined in what trace and using that to clear existing traces or such...

Reply:
So you insert additional interrupt check just to check if the function was blacklisted at run-time and invalidate the current trace. How do you do this?
- why can't you do this "black work" in ``jit_observer_fcall_begin()``?
- you still miss the case when trace ir returned to blacklisted function.

Reply:
I think, the thing you are trying to achieve is - "lazy deoptimization".
There were some related V8 papers/presentations.

Reply:
Yeah, but actually doing it properly is fully overkill for our needs :-)